osu.Game/Online/Spectator/SpectatorStreamingClient.cs

```diff
@@ -292,7 +298,10 @@ protected override void Update()
         {
             base.Update();
 
-            if (pendingFrames.Count > 0 && Time.Current - lastSendTime > TIME_BETWEEN_SENDS)
+            double timeSinceLastSend = Time.Current - lastSendTime;
+
+            if ((pendingFrames.Count > 0 && timeSinceLastSend > MIN_TIME_BETWEEN_SENDS)
+                || (currentScore != null && timeSinceLastSend > MAX_TIME_BETWEEN_SENDS))
```

Comment:
On revisiting this one, this check is causing empty frames to be sent outside of gameplay. `currentScore` is not actually cleared in `endPlaying` - and doing that can actually trip an assertion in `purgePendingFrames()`.

The following seems to work okay:

```suggestion
                || (isPlaying && timeSinceLastSend > MAX_TIME_BETWEEN_SENDS))
```